test/GPv2Encoding.test.ts

```diff
@@ -327,6 +329,124 @@ describe("GPv2Encoding", () => {
         .be.reverted;
     });
 
+    it("should verify EIP-1271 contract signatures by returning owner", async () => {
+      const artifact = await artifacts.readArtifact("EIP1271Verifier");
+      const verifier = await waffle.deployMockContract(deployer, artifact.abi);
+
+      const message = eip1271Message(testDomain, sampleOrder);
+      const eip1271Signature = "0x031337";
+      await verifier.mock.isValidSignature
+        .withArgs(message, eip1271Signature)
+        .returns(EIP1271_MAGICVALUE);
+
+      const encoder = new SettlementEncoder(testDomain);
+      await encoder.encodeContractTrade(
+        sampleOrder,
+        verifier.address,
+        eip1271Signature,
+      );
+
+      const { trades } = await encoding.decodeTradesTest(
+        encoder.tokens,
+        encoder.encodedTrades,
+      );
+
+      expect(trades.length).to.equal(1);
+      const { owner } = decodeTrade(trades[0]);
+      expect(owner).to.equal(verifier.address);
+    });
+
+    it("should revert on an invalid EIP-1271 signature", async () => {
+      const message = eip1271Message(testDomain, sampleOrder);
+      const eip1271Signature = "0x031337";
+
+      const artifact = await artifacts.readArtifact("EIP1271Verifier");
+      const verifier1 = await waffle.deployMockContract(deployer, artifact.abi);
+
+      await verifier1.mock.isValidSignature
+        .withArgs(message, eip1271Signature)
+        .returns(Number.parseInt(EIP1271_MAGICVALUE) - 1);
```

Comment:
Technically, I suppose this return value can be anything other than the magic number. Right?

Reply:
petition to change it to `0xbaadc0d3` :nerd_face: